pkg/collector/corechecks/servicediscovery/module/comm_test.go

```diff
@@ -124,9 +133,78 @@ func TestShouldIgnoreComm(t *testing.T) {
 			require.EventuallyWithT(t, func(collect *assert.CollectT) {
 				proc, err := customNewProcess(int32(cmd.Process.Pid))
 				require.NoError(t, err)
-				ignore := shouldIgnoreComm(proc)
+
+				ignore := discovery.shouldIgnoreComm(proc)
 				require.Equal(t, test.ignore, ignore)
-			}, 30*time.Second, 100*time.Millisecond)
+			}, 10*time.Second, 100*time.Millisecond)
```

Comment:
1. what's the meaning of using require.Eventually here? Do we expect the result to change over time?
2. you cannot use `require.*` methods inside `require.Eventually*`. Try to see what will happen if one of the internal calls to `require.*` fails

Reply:
cmd.Start() does not guarantee immediate process start, but eventually process will run and will be discovered. I've changed  "require" to "assert" in latest commit

Reply:
I've refactored this unit test